### DISCUSSION TITLE
Deploying static build to GitHub creates public folder for images unnecessarily

### DISCUSSION THREAD

**Discussion by landed1** [vercel/next.js#73279] [Help] [answered]
### Summary

I see that my default 'out' folder builds fine on my local machine. So I have
out/
---img/images

but when I deploy and build on GitHub it creates a public folder with images inside that

GitHub/
---public/
------img/images

So this breaks my images on GitHub Pages. I can't see how this is happening. I used the default yarn build script.

Any ideas? Yes I could use an environment variable to add this 'public/' to all image url's but it feels wrong and lead to future bugs.

**Comment by icyJoseph** [comment]
Don't you have to set a `gh-pages` or whatever name, branch where the `out` folder is to be deployed as the repository contents, so to speak?

- https://dev.to/lico/nextjs-deploy-as-static-site-using-github-pages-3bhm

**Reply by landed1** [reply]
Thank you for your help. In my case GitHub pages detected that it was a next app and suggests an action with a default YAML build file. I chose this route and kept all the defaults. It worked. I was thinking that the github repository was the 'build' long day.

**Comment by landed1** [comment, marked as answer]
I think I am getting confused that the files I see in github are not what is built so then the question maybe just that I am using an absolute file path for the images.
Yes this was it. Silly me. So you need to make paths relative.
